src/assistant.js

```diff
@@ -0,0 +1,341 @@
+  if (isWeekend || outsideBusinessHours) {
+    actions = [
+      {
+        say: `${message} Please hold while we connect you with an operator as you are calling outside of regular business hours.`,
+      },
+      {
+        handoff: {
+          channel: 'voice',
+          uri: `${baseUrl}/operator-webhook`,
+          method: 'POST',
+        },
+      },
+    ];
+  } else {
+    actions = [
+      {
+        say: message,
+      },
+      {
+        redirect: 'task://main-menu',
+      },
+    ];
+  }
+  return actions;
+}
+
+/**
+ * Gets or creates a new assistant for the IVR Tutorial
+ * @param {string} name
+ * @param {string} baseUrl
+ * @return {Promise<AssistantInstance>}
+ */
+async function updateOrCreateAssistant(name, baseUrl) {
+  const uniqueName = slugify(name);
+  const assistants = await client.autopilot.assistants.list();
+
+  let assistant = assistants.find(
+    assistant => assistant.uniqueName === uniqueName
+  );
+
+  if (assistant === undefined) {
+    console.log(`Create assistant: "${name}"`);
+    assistant = await client.autopilot.assistants.create({
+      friendlyName: name,
+      uniqueName: uniqueName,
+      defaults: {
+        defaults: {
+          assistant_initiation: `${baseUrl}/autopilot/greeting`,
+          fallback: `task://main-menu`,
+        },
+      },
+      styleSheet: {
+        style_sheet: {
+          voice: {
+            say_voice: 'Polly.Matthew',
+          },
+        },
+      },
+    });
+  } else {
+    console.log(`Update assistant: "${name}"`);
+    await assistant.update({
+      defaults: {
+        defaults: {
+          assistant_initiation: `${baseUrl}/autopilot/greeting`,
+          fallback: `task://main-menu`,
+        },
+      },
+      styleSheet: {
+        style_sheet: {
+          voice: {
+            say_voice: 'Polly.Matthew',
+          },
+        },
+      },
+    });
+  }
+
+  return assistant;
+}
+
+/**
+ * Links a phone number to an autopilot assistant
+ * @param {AssistantInstance} assistant
+ * @param {string} accountSid
+ * @param {string} phoneNumber
+ * @return {Promise<void>}
+ */
+async function linkAssistantToPhoneNumber(assistant, accountSid, phoneNumber) {
+  console.log('Link phone number to assistant');
+  const incomingPhoneNumbers = await client.api.incomingPhoneNumbers.list();
+  const incomingPhoneNumber = incomingPhoneNumbers.find(
+    incomingPhoneNumbers => incomingPhoneNumbers.phoneNumber === phoneNumber
+  );
+  await incomingPhoneNumber.update({
+    voiceMethod: 'POST',
+    voiceUrl: `https://channels.autopilot.twilio.com/v1/${accountSid}/${assistant.sid}/twilio-voice`,
+  });
+}
+
+/**
+ * Create or update the Main Menu task for the assistant
+ * @param {AssistantInstance} assistant
+ * @param {TaskInstance[]} tasks
+ * @param {string} baseUrl
+ * @return {Promise<TaskInstance>}
+ */
+async function createOrUpdateMainMenuTask(assistant, tasks, baseUrl) {
+  let task = tasks.find(task => task.uniqueName === 'main-menu');
```

Comment:
Same here, as with `assistant`, we should avoid mutating objects as much as possible.